### ISSUE TITLE
Regression: Argument of type ... is not assignable to parameter of type ...

### ISSUE THREAD

**Issue by asfernandes** [microsoft/TypeScript#29505]
This were working without error in TypeScript 3.0.1 but is not working on 3.2.4.

The `select` function is intended to accept only property names of type string in the last argument, but in 3.2.4 it does not accept any string.

It gives error: `[ts] Argument of type '"value"' is not assignable to parameter of type 'XX extends XX ? "value" : never'. [2345]`

```ts
export type FilterPropsByType<T, TT> = {
	[K in keyof T]: T[K] extends TT ? K : never
}[keyof T];

function select<
	T extends string | number,
	TList extends object,
	TValueProp extends FilterPropsByType<TList, T>
>(
	property: T, list: TList[], valueProp: TValueProp
)
{
	//
}

export function func<XX extends string>(x: XX, tipos: { value: XX }[]) {
	select(x, tipos, 'value');
}
```

**Comment by jack-williams** [comment]
I think this is a bug, but I'm not 100%. This will probably need @weswigham or @ahejlsberg to answer definitively. 

I think this is a bug because wrapping the conditional type in a 1-tuple fixes the error, but it shouldn't really make a difference because the conditional type was never distributive in the first place. This works:
```ts
export type FilterPropsByType<T, TT> = {
	[K in keyof T]: [T[K]] extends [TT] ? K : never
}[keyof T];
```

Specifically, the code is skipping past this check because both the check and extends type are instantiable.
```ts
// We attempt to resolve the conditional type only when the check and extends types are non-generic
if (!checkTypeInstantiable && !maybeTypeOfKind(inferredExtendsType, TypeFlags.Instantiable | TypeFlags.GenericMappedType)) {
    if (inferredExtendsType.flags & TypeFlags.AnyOrUnknown) {
        return instantiateType(root.trueType, mapper);
    }
```
and then dropping down into deferral:
```ts
// Return a deferred type for a check that is neither definitely true nor definitely false
const erasedCheckType = getActualTypeVariable(checkType);
const result = <ConditionalType>createType(TypeFlags.Conditional);
```

Checking for identity before deferral does fix the issue, but I'm not sure it's the right thing to do:
```ts
if (isTypeIdenticalTo(checkType, extendsType)) {
    return instantiateType(root.trueType, combinedMapper || mapper);
}
// Return a deferred type for a check that is neither definitely true nor definitely false
// ….
```

**Comment by RyanCavanaugh** [comment]
@weswigham tentatively tagging this as a bug because it is indeed a regression relative to 3.1

**Comment by asfernandes** [comment]
This bug was target for 3.5.0 milestone but it's still present in 3.5.1. I just cannot update TypeScript since 3.0.1 in my project.

**Comment by asfernandes** [comment]
I just tested 3.6.0-dev.20190720 and the bug is still present.

**Comment by weswigham** [comment]
`Fix Available` means there's a PR with the fix open, but it has yet to be merged (for whatever reason) - the issue will be closed once the fix is merged.

**Comment by RyanCavanaugh** [comment]
The linked PR is pretty wide-reaching so I'm slightly hesitant to call it a definitive fix for the issue here
